tests/components/hdmi_cec/test_media_player.py

```diff
@@ -0,0 +1,493 @@
+        ),
+        (TYPE_RECORDER, {"power_status": POWER_ON, "status": STATUS_STOP}, STATE_IDLE),
+        (
+            TYPE_RECORDER,
+            {"power_status": POWER_ON, "status": STATUS_STILL},
+            STATE_PAUSED,
+        ),
+        (TYPE_RECORDER, {"power_status": POWER_ON, "status": None}, STATE_UNKNOWN),
+    ],
+)
+async def test_update_state(
+    hass, create_hdmi_network, create_cec_entity, type_id, update_data, expected_state
+):
+    """Test state updates work as expected."""
+    hdmi_network = await create_hdmi_network({"platform": "media_player"})
+    mock_hdmi_device = MockHDMIDevice(logical_address=3, type=type_id)
+    await create_cec_entity(hdmi_network, mock_hdmi_device)
+
+    for att, val in update_data.items():
+        setattr(mock_hdmi_device, att, val)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("media_player.hdmi_3")
+    assert state.state == expected_state
+
+
+@pytest.mark.parametrize(
+    "data,expected_state",
+    [
+        ({"power_status": POWER_OFF}, STATE_OFF),
+        ({"power_status": 3}, STATE_OFF),
+        ({"power_status": POWER_ON, "type": TYPE_TV}, STATE_ON),
+        ({"power_status": 4, "type": TYPE_TV}, STATE_ON),
+        ({"power_status": POWER_ON, "type": TYPE_TV, "status": STATUS_PLAY}, STATE_ON),
+        (
+            {"power_status": POWER_OFF, "type": TYPE_RECORDER, "status": STATUS_PLAY},
+            STATE_OFF,
+        ),
+        (
+            {"power_status": POWER_ON, "type": TYPE_RECORDER, "status": STATUS_PLAY},
+            STATE_PLAYING,
+        ),
+        (
+            {"power_status": POWER_ON, "type": TYPE_RECORDER, "status": STATUS_STOP},
+            STATE_IDLE,
+        ),
+        (
+            {"power_status": POWER_ON, "type": TYPE_RECORDER, "status": STATUS_STILL},
+            STATE_PAUSED,
+        ),
+        (
+            {"power_status": POWER_ON, "type": TYPE_RECORDER, "status": None},
+            STATE_UNKNOWN,
+        ),
+    ],
+)
+async def test_starting_state(
+    hass, create_hdmi_network, create_cec_entity, data, expected_state
+):
+    """Test starting states are set as expected."""
+    hdmi_network = await create_hdmi_network({"platform": "media_player"})
+    mock_hdmi_device = MockHDMIDevice(logical_address=3, **data)
+    await create_cec_entity(hdmi_network, mock_hdmi_device)
+    state = hass.states.get("media_player.hdmi_3")
+    assert state.state == expected_state
+
+
+@pytest.mark.xfail(
+    reason="The code only sets the state to unavailable, doesn't set the `_attr_available` to false."
+)
+async def test_unavailable_status(hass, create_hdmi_network, create_cec_entity):
+    """Test entity goes into unavailable status when expected."""
+    hdmi_network = await create_hdmi_network({"platform": "media_player"})
+    mock_hdmi_device = MockHDMIDevice(logical_address=3)
+    await create_cec_entity(hdmi_network, mock_hdmi_device)
+
+    hass.bus.async_fire(EVENT_HDMI_CEC_UNAVAILABLE)
```

Comment:
For the future: The integration shouldn't fire an event to make the entities go unavailable but use our dispatch helper to signal the entities. Alternatively implement our DataUpdateCoordinator helper which also works in push mode.